packages/picker/src/InteractionController.ts

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2024 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+    ReactiveController,
+    TemplateResult,
+} from '@spectrum-web-components/base';
+import { AbstractOverlay } from '@spectrum-web-components/overlay/src/AbstractOverlay';
+import { Overlay } from '@spectrum-web-components/overlay/src/Overlay.js';
+import { PickerBase } from './Picker.js';
+
+export enum InteractionTypes {
+    'desktop',
+    'mobile',
+}
+
+export class InteractionController implements ReactiveController {
+    abortController!: AbortController;
+
+    public preventNextToggle: 'no' | 'maybe' | 'yes' = 'no';
```

Comment:
`maybe` is required to capture this case here
```js
protected handleButtonFocus(event: FocusEvent): void {
        // When focus comes from a pointer event, and the related target is the Menu,
        // we don't want to reopen the Menu.
        if (
            this.preventNextToggle === 'maybe' &&
            event.relatedTarget === this.optionsMenu
        ) {
            this.preventNextToggle = 'yes';
        }
    }
 ```

Reply:
Yes, but what I meant is that this could have been achieved by using a boolean everywhere else and simply toggling the value of `preventNextToggle` in the `handleButtonFocus`:

```ts
protected handleButtonFocus(event: FocusEvent): void {
        // When focus comes from a pointer event, and the related target is the Menu,
        // we don't want to reopen the Menu.
        if (
            this.preventNextToggle &&
            event.relatedTarget !== this.host.optionsMenu
        ) {
            this.preventNextToggle = false;
        }
    }
```
But this is nitpicking - let this comment be non-blocking.